### PR TITLE
merge sorted arrays

### PR DESCRIPTION
**Pull request by jamestitus299**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/arrays/merge_two_sorted_arrays.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    Examples:
+        >>> merge_two_sorted_arrays([1,5,7], [0,4,9])
+        [0, 1, 4, 5, 7, 9]
+
+        >>> merge_two_sorted_arrays([-11,5,45], [0,5,9])
+        [-11, 0, 5, 5, 9, 45]
+    """
+
+    if not nums1 and not nums2:
+        raise ValueError("Both input arrays are empty.")
```
Comment (cclauss) [inline data_structures/arrays/merge_two_sorted_arrays.py:26]: We need to validate that both lists are sorted.

File: data_structures/arrays/merge_two_sorted_arrays.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        nums2: The second array.
+
+    Returns:
+        The merged sorted array.
+
+    Examples:
+        >>> merge_two_sorted_arrays([1,5,7], [0,4,9])
+        [0, 1, 4, 5, 7, 9]
+
+        >>> merge_two_sorted_arrays([-11,5,45], [0,5,9])
```
Comment (cclauss) [inline data_structures/arrays/merge_two_sorted_arrays.py:21]:
```suggestion
        >>> merge_two_sorted_arrays([1, 5, 7], [0, 4, 9])
        [0, 1, 4, 5, 7, 9]

        >>> merge_two_sorted_arrays([-11, 5, 45], [0, 5, 9])
```

**Comment by dkshsingh** [comment]
Merge Sort array Implementation code in Python:

def mergesort(a):
    if len(a) > 1:
        mid = len(a) // 2
        left_half = a[:mid]
        right_half = a[mid:]

        mergesort(left_half)
        mergesort(right_half)

        i = j = k = 0

        while i < len(left_half) and j < len(right_half):
            if left_half[i] < right_half[j]:
                a[k] = left_half[i]
                i += 1
            else:
                a[k] = right_half[j]
                j += 1
            k += 1

        while i < len(left_half):
            a[k] = left_half[i]
            i += 1
            k += 1

        while j < len(right_half):
            a[k] = right_half[j]
            j += 1
            k += 1

def merge_sort(a):
    mergesort(a)

if __name__ == "__main__":
    a = [int(x) for x in input("Enter array elements separated by spaces: ").split()]
    
    merge_sort(a)
    
    print("\nSorted array is:", a)

File: data_structures/arrays/merge_two_sorted_arrays.py (hunk 2 above)
Comment (jamestitus299) [reply data_structures/arrays/merge_two_sorted_arrays.py]: Done!

**Comment by cclauss** [comment]
We already have two merge sorts: https://github.com/TheAlgorithms/Python/tree/master/sorts

Thank you for your contribution to TheAlgorithms/Python but this has been a __very busy HacktoberFest__ so we will be closing all new pull requests until the end of October so we can properly review approximately 200 open pull requests.
### We hope you will try to contribute again after October 31st.

**Pull request closed without merging**